src/uct/ib/mlx5/dc/dc_mlx5.inl

```diff
@@ -41,17 +41,29 @@ uct_dc_mlx5_get_arbiter_params(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep,
 }
 
 static UCS_F_ALWAYS_INLINE void
-uct_dc_mlx5_ep_schedule(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
+uct_dc_mlx5_ep_push_pending_req(ucs_arbiter_group_t *group,
+                                uct_pending_req_t *r, int push_to_head)
 {
-    if (ep->dci == UCT_DC_MLX5_EP_NO_DCI) {
-        /* no dci:
-         * Do not grab dci here. Instead put the group on dci allocation
-         * arbiter. This way we can assure fairness between all eps waiting for
-         * dci allocation. Relevant for dcs and dcs_quota policies.
-         */
-        uct_dc_mlx5_iface_schedule_dci_alloc(iface, ep);
+    if (push_to_head) {
+        uct_pending_req_arb_group_push_head(group, r);
     } else {
-        uct_dc_mlx5_iface_dci_sched_tx(iface, ep);
+        uct_pending_req_arb_group_push(group, r);
+    }
+}
+
+static UCS_F_ALWAYS_INLINE void
+uct_dc_mlx5_ep_pending_common_shared(uct_dc_mlx5_iface_t *iface,
+                                     uct_dc_mlx5_ep_t *ep, uct_pending_req_t *r,
+                                     unsigned flags, int push_to_head,
+                                     int schedule)
+{
+    ucs_arbiter_group_t *group = uct_dc_mlx5_ep_shared_arb_group(iface, ep);
+
+    uct_dc_mlx5_pending_req_priv(r)->ep = ep;
```

Comment:
i'd also move the UCS_STATIC_ASSERT from uct_dc_mlx5_ep_pending_common to here, since we use uct_dc_mlx5_pending_req_priv only in this function now